Checkout/Sample/Source/Card Tokenisation/Card Details/ExpirationDatePicker.swift

```diff
@@ -0,0 +1,155 @@
+//
+//  ExpirationDatePicker.swift
+//
+//
+//  Created by Harry Brown on 08/02/2022.
+//
+
+import UIKit
+
+class ExpirationDatePicker: UIPickerView {
+  private static var secondsInYear: TimeInterval = 31536000
+
+  struct ViewModel {
+    var onDateChanged: ((_ month: Int, _ year: Int) -> Void)?
+  }
+
+  var viewModel: ViewModel?
+
+  private var months: [String] = []
+  private var years: [Int] = []
```

Comment:
These feel like lets? 

For months, you can use something like 
```
private let months = 1...12.map { formatter.string(from: 0) } // If you use a numberFormatter with formatter.minimumIntegerDigits = 2
private let months = 1...12.map { String(format: "%02d", $0) }
```

Similarly for year, I am not sure you need to create dates from minimum to maximum, then break them down and fetch the years. Something like this may also work and save you some properties
```
let years = 0...20.map { currentYear + $0 }
```

Reply:
Yeah makes sense, I've done a bit of cleanup here. the ExpirationDatePicker was originally from the old Frames UI. I originally put minimal work into refactoring because it would only be used in the sample app